shared/src/ledger/eth_bridge/bridge_pool_vp.rs

```diff
@@ -169,36 +209,62 @@ where
             return Ok(false);
         }
 
-        // check that gas fees were put into escrow
-
-        // check that the correct amount was deducted from the fee payer
-        if let Some(SignedAmount::Negative(amount)) =
-            self.account_balance_delta(&transfer.gas_fee.payer)
-        {
-            if amount != transfer.gas_fee.amount {
-                return Ok(false);
-            }
+        // if we are going to mint wNam on Ethereum, the appropriate
+        // amount of Nam must be escrowed in the Ethereum bridge VP's storage.
+        // TODO: We should look this address up from storage
+        if transfer.transfer.asset == wnam() {
+            // check that correct amount of Nam was put into escrow.
+            return if transfer.gas_fee.payer == transfer.transfer.sender {
+                if !self.check_nam_escrowed(
+                    &transfer.gas_fee.payer,
+                    &BRIDGE_POOL_ADDRESS,
+                    transfer.gas_fee.amount + transfer.transfer.amount,
```

Comment:
might be worth adding overflow checks here. in `release` builds, these are simply ignored by the compiler, just like in C/C++.